### ISSUE TITLE
Pass kwargs to find_matches?

### ISSUE THREAD

**Issue by lilyminium** [openforcefield/openff-interchange#347]
In various store_matches methods, parameter_handler.find_matches is called with default arguments. I want to do bad hacky things with Interchange and it'd be lovely if kwargs could be passed in. If you want users not to do bad hacky things, it's possibly a good idea to not do this.

**Comment by mattwthompson** [comment]
Just so I understand, the idea is something like this?

```python3
    def store_matches(
        self,
        parameter_handler: ParameterHandler,
        topology: Union["Topology", "_OFFBioTop"],
        **kwargs,
    ) -> None:
        ...
        matches = parameter_handler.find_matches(topology, **kwargs)
```

I don't have a strong opinion on this - I certainly want to enable powerusers like yourself to do creative things. But it is a genie that's hard to put back in the bottle if it's used for unsound things.

**Comment by lilyminium** [comment]
>  I certainly want to enable powerusers like yourself to do creative things. But it is a genie that's hard to put back in the bottle if it's used for unsound things.

Yes, definitely -- although I think with `find_matches` at least, it only acknowledges `unique` and ignores other kwargs passed in.

I don't have strong opinions on this either way, as anyone who's determined enough can just copy and paste the existing method to modify necessary lines. It's more to do with how extensible you think Interchange/the toolkit should be.

**Comment by mattwthompson** [comment]
> I think with `find_matches` at least, it only acknowledges `unique` and ignores other kwargs passed in.

This is what I saw as well - but I suspect it's a place the toolkit might expand its flexibility in the future, so I'm inclined to support passing options between them. It's hard to lock users out of doing this with Python code, and I'm not sure I'd want to; in this case having it as an "opt-in" feature would at least ensure that breakages aren't likely to happen to users by accident (short of copying code, etc.).

**Comment by mattwthompson** [comment]
@lilyminium my opinion on `**kwargs` has changed over the years - is this something that would still be useful enough for you and your teams' work that you'd advocate for it to be included?
